src/internal/analytics/components/analytics-funnel.tsx

```diff
@@ -52,6 +52,16 @@ export const AnalyticsFunnel = (props: AnalyticsFunnelProps) => {
   return <InnerAnalyticsFunnel {...props} />;
 };
 
+const funnelCancelled = ({ funnelInteractionId }: { funnelInteractionId: string }) => {
+  FunnelMetrics.funnelCancelled({ funnelInteractionId });
+  document.dispatchEvent(new Event('cloudscape-creation-flow-done'));
+};
+
+const funnelComplete = ({ funnelInteractionId }: { funnelInteractionId: string }) => {
+  FunnelMetrics.funnelComplete({ funnelInteractionId });
+  document.dispatchEvent(new Event('cloudscape-creation-flow-done'));
```

Comment:
I recommend setting the event name in a const instead. This way, we'll only have to change one place if the constant value needs to be changed in the future.